cpp/CMakeLists.txt

```diff
@@ -43,8 +43,9 @@ if(POLICY CMP0054)
   cmake_policy(SET CMP0054 NEW)
 endif()
 
+# The SO version should also be the ABI version
 set(ARROW_SO_VERSION "0")
-set(ARROW_ABI_VERSION "${ARROW_SO_VERSION}.0.0")
```

Comment:
We're using `ARROW_ABI_VERSION` in `cpp/src/arrow/arrow.pc.in`.
Please keep the name.

Reply:
"ABI version" is really the wrong name for this. The ABI version doesn't change every patch release.

Reply:
OK.
We have similar codes for Plasma in cpp/src/CMakeLists.txt and cpp/src/plasma/plasma.pc.in. It's better that we change them too.

Reply:
One more thing.
I did the same thing (`PARQUET_ABI_VERSION` was introduced) for parquet-cpp. It may be better that we change them too for consistency. We have a code that depends on the in `cpp/cmake_modules/FindParquet.cmake`.

Reply:
Note that `PLASMA_VERSION` is used in `plasma.pc.in` but never defined anywhere...